book/src/contributing/architecture.md

```diff
@@ -0,0 +1,84 @@
+# Architecture
+
+Hax is a software pipeline designed to transform Rust code into various formal verification backends such as **F\***, **Coq**, **ProVerif**, and **EasyCrypt**. It comprises two main components:
+
+1. **The Frontend** (written in Rust)
+2. **The Engine** (written in OCaml)
+
+## The Frontend (Rust)
+
+The frontend is responsible for extracting and exporting Rust code's abstract syntax trees (ASTs) in a format suitable for processing by the engine (or by other tools).
```

Comment:
You start describing pieces below without putting them into relation. The section up here should tell what's coming and how they are related. The same goes for the top level section.

It should also contain what's public things and what are internal things.

Reply:
Hm, I don't see a new paragraph here.

Reply:
Sorry, messed up with the history, I re-commited, that's commit afff4b381b6454b5d53521d4c03729ac007dd8c7 now